wcfsetup/install/files/lib/system/service/worker/Encryption.class.php

```diff
@@ -0,0 +1,194 @@
+            'aes-128-gcm',
+            $cek,
+            OPENSSL_RAW_DATA,
+            $nonce,
+            $tag
+        );
+
+        if ($serviceWorker->contentEncoding === ServiceWorker::CONTENT_ENCODING_AESGCM) {
+            $headers['Encryption'] = 'salt=' . Base64UrlSafe::encode($salt);
+            $headers['Crypto-Key'] = 'dh=' . Base64UrlSafe::encode($newPublicKey);
+        }
+        $record = $encryptedText . $tag;
+
+        return Encryption::getEncryptionContentCodingHeader(
+            $serviceWorker->contentEncoding,
+            \mb_strlen($record, '8bit'),
+            $salt,
+            $newPublicKey
+        ) . $record;
+    }
+
+    private static function getSharedSecret(JWK $publicKey, #[\SensitiveParameter] JWK $privateKey): string
+    {
+        $result = \openssl_pkey_derive(
+            ECKey::convertPublicKeyToPEM($publicKey),
+            ECKey::convertPrivateKeyToPEM($privateKey),
+            256
+        );
+        \assert($result);
+        return \str_pad($result, 32, "\x00", STR_PAD_LEFT);
+    }
+
+    private static function addPadding(#[\SensitiveParameter] string $payload, string $contentEncoding): string
+    {
+        $length = \mb_strlen($payload, '8bit');
+        $paddingLength = ServiceWorkerHandler::MAX_PAYLOAD_LENGTH - $length;
+
+        if ($contentEncoding === ServiceWorker::CONTENT_ENCODING_AES128GCM) {
+            return \str_pad($payload . "\x02", $paddingLength + $length, "\x00", STR_PAD_RIGHT);
+        } elseif ($contentEncoding === ServiceWorker::CONTENT_ENCODING_AESGCM) {
+            return \pack('n*', $paddingLength) . \str_pad($payload, $paddingLength + $length, "\x00", STR_PAD_LEFT);
+        } else {
+            throw new \RuntimeException('Unknown content encoding "' . $contentEncoding . '"');
+        }
+    }
+
+    private static function createInfo(string $type, ?string $context, string $contentEncoding)
+    {
+        if ($contentEncoding === ServiceWorker::CONTENT_ENCODING_AESGCM) {
+            \assert($context !== null);
+            \assert(\mb_strlen($context, '8bit') === 135);
+
+            return 'Content-Encoding: ' . $type . "\x00" . Encryption::CURVE_ALGORITHM . $context;
+        } elseif ($contentEncoding === ServiceWorker::CONTENT_ENCODING_AES128GCM) {
+            return 'Content-Encoding: ' . $type . "\x00";
+        } else {
+            throw new \RuntimeException('Unknown content encoding "' . $contentEncoding . '"');
+        }
+    }
+
+    private static function getEncryptionContentCodingHeader(
+        string $contentEncoding,
+        int $length,
+        string $salt,
+        string $publicKey
+    ): string {
+        if ($contentEncoding === ServiceWorker::CONTENT_ENCODING_AES128GCM) {
+            /** {@link https://datatracker.ietf.org/doc/html/rfc8188#section-2.1} */
+            return $salt
+                . \pack('N*', $length)
+                . \pack('C*', \mb_strlen($publicKey, '8bit'))
+                . $publicKey;
```

Comment:
```suggestion
            return \pack(
                'A*NCA*,
                $salt,
                $length,
                \strlen($publicKey)
                $publicKey
            );
```

Just `\strlen()` is fine with PHP 8.0+.